lib/common/options.c

```diff
@@ -289,8 +290,16 @@ pcmk__env_option(const char *option)
     char env_name[NAME_MAX];
     const char *value = NULL;
 
+    CRM_CHECK(!pcmk__str_empty(option), return NULL);
+
     for (int i = 0; i < (sizeof(prefixes) / sizeof(char *)); i++) {
-        snprintf(env_name, NAME_MAX, "%s%s", prefixes[i], option);
+        int rv = snprintf(env_name, NAME_MAX, "%s%s", prefixes[i], option);
+
+        if (rv >= sizeof(env_name)) {
```

Comment:
I considered also checking for a negative `rv` just in case. Still open to it. It's reasonable, but the logical conclusion is to add a check and error handler everywhere we call `snprintf()`... and there are a lot of them.

Reply:
Yes actually, that is a goal. Taking care of these is worthwhile but don't worry about the rest. I plan to replace the offset+= calls with GString or something along those lines.

Reply:
Done.